spec/active_model/composite_command_spec.rb

```diff
@@ -110,5 +110,65 @@ def exception_command
         end
       end
     end
+
+    context "when composite command uses with transaction helper" do
+      let(:command) { WithTransactionCompositeCommand.new([composite]) }
+      let(:composite) { CompositeCommand.new }
+
+      before "stablish in memory database connection to work with transactions" do
+        ActiveRecord::Base.establish_connection(adapter: "sqlite3", database: ":memory:")
+      end
+
+      it "executes block inside a transaction" do
+        expect(ActiveRecord::Base).to receive(:transaction)
+        command.call
+      end
+
+      context "on success" do
+        let(:composite) { CompositeCommand.new([success_command]) }
+
+        before do
+          command.call
+        end
+
+        let(:subcommands) { [success_composite] }
+
+        it { is_expected.to be_success }
+
+        it "has a result" do
+          expect(command.result).to eq(:result)
+        end
+      end
+
+      context "on failure" do
+        before do
+          command.call
+        end
+
+        let(:composite) { CompositeCommand.new([failure_command]) }
+
+        it "adds errors" do
+          expect(errors[:base]).to include(/failure/)
+        end
+
+        it "has a result" do
+          expect(command.result).to eq(:result)
```

Comment:
This was confusing to me, I would expected to be `nil` on this case:

When we raise the exception due to failure [here](https://github.com/goldstar/active_model_command/blob/master/lib/active_model/composite_command.rb#L32) 

The composite command behaves different:

##### Without using `with_transaction` helper
It will rescue the exception on the CompositeCommand [here](https://github.com/goldstar/active_model_command/blob/master/lib/active_model/composite_command.rb#L18) so after setting result to `nil` [here](https://github.com/goldstar/active_model_command/blob/master/lib/active_model/composite_command.rb#L25), the result will be `nil` [here](https://github.com/goldstar/active_model_command/blob/master/lib/active_model/command.rb#L44)

#####  Using `with_transaction` helper
It will rescue the exception on the helper itself (inside the `WithTransactionCompositeCommand` we have)
(https://github.com/goldstar/active_model_command/blob/master/lib/active_model/composite_command.rb#L18) so after setting result to `nil` [here] the execution of the user's command continues, so the result of the command will be https://github.com/goldstar/active_model_command/pull/11/files#diff-cf609e669b33ca80ea1d1d5e05795850f7259efc3f6047c44ca9c7b8aea0ca74R18 and will override the result that we already set to `nil`.